File: GTTS/INL.UnitLibraryService.Models/SaveUnit_Param.cs
using System;
using System.Collections.Generic;

namespace INL.UnitLibraryService.Models
{
    public class SaveUnit_Param : ISaveUnit_Param
    {
        public long? UnitID { get; set; }
        public long? UnitParentID { get; set; }
        public int CountryID { get; set; }
        public long? UnitLocationID { get; set; } 
        public int? ConsularDistrictID { get; set; }
        public string UnitName { get; set; }
        public string UnitNameEnglish { get; set; }
        public bool? IsMainAgency { get; set; }
        public long? UnitMainAgencyID { get; set; }
        public string UnitAcronym { get; set; }
        public string UnitGenID { get; set; }
        public List<SaveUnitAlias_Item> UnitAlias { get; set; }
        public int UnitTypeID { get; set; }
        public int? GovtLevelID { get; set; }
        public int? UnitLevelID { get; set; }
        public byte VettingBatchTypeID { get; set; }
        public int VettingActivityTypeID { get; set; }
        public int? ReportingTypeID { get; set; }
        public long? UnitHeadPersonID { get; set; }
        public string UnitHeadJobTitle { get; set; }
        public int? UnitHeadRankID { get; set; }
        public long? HQLocationID { get; set; }
        public string POCName { get; set; }
        public string POCEmailAddress { get; set; }
        public string POCTelephone { get; set; }
        public string VettingPrefix { get; set; }
        public bool HasDutyToInform { get; set; }
        public bool IsLocked { get; set; }
        public bool IsActive { get; set; }
        public int ModifiedByAppUserID { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public SaveUnitLocation_Item UnitLocation { get; set; }
        public SaveUnitLocation_Item HQLocation { get; set; }
        public SaveUnitCommander_Item Commander { get; set; }
    }
}

File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.UnitLibraryService.Models/save-unit_param.ts


import { SaveUnitAlias_Item } from './save-unit-alias_item';
import { SaveUnitLocation_Item } from './save-unit-location_item';
import { SaveUnitCommander_Item } from './save-unit-commander_item';
import { ISaveUnit_Param } from './isave-unit_param';

export class SaveUnit_Param implements ISaveUnit_Param {
  
	public UnitID?: number;
	public UnitParentID?: number;
	public CountryID: number = 0;
	public UnitLocationID?: number;
	public ConsularDistrictID?: number;
	public UnitName: string = "";
	public UnitNameEnglish: string = "";
	public IsMainAgency?: boolean;
	public UnitMainAgencyID?: number;
	public UnitAcronym: string = "";
	public UnitGenID: string = "";
	public UnitAlias?: SaveUnitAlias_Item[];
	public UnitTypeID: number = 0;
	public GovtLevelID?: number;
	public UnitLevelID?: number;
	public VettingBatchTypeID: number = 0;
	public VettingActivityTypeID: number = 0;
	public ReportingTypeID?: number;
	public UnitHeadPersonID?: number;
	public UnitHeadJobTitle: string = "";
	public UnitHeadRankID?: number;
	public HQLocationID?: number;
	public POCName: string = "";
	public POCEmailAddress: string = "";
	public POCTelephone: string = "";
	public VettingPrefix: string = "";
	public HasDutyToInform: boolean = false;
	public IsLocked: boolean = false;
	public IsActive: boolean = false;
	public ModifiedDate?: Date;
	public UnitLocation?: SaveUnitLocation_Item;
	public HQLocation?: SaveUnitLocation_Item;
	public Commander?: SaveUnitCommander_Item;
  
}


